inheritance/node-api/inheritance.c

```diff
@@ -0,0 +1,226 @@
+                      "subclass_superclass_chained_method");
+  return NULL;
+}
+
+static napi_value
+subclass_superclass_chained_static_method(napi_env env,
+                                          napi_callback_info info) {
+  superclass_chained_static_method(env, info);
+  SET_TARGET_PROPERTY(env, false, info,
+                      "subclass_superclass_chained_static_method");
+  return NULL;
+}
+
+static napi_value Init(napi_env env, napi_value exports) {
+  napi_value superclass, subclass;
+
+  napi_property_descriptor superclass_properties[] = {
+      {"superclassMethod", NULL, superclass_method, NULL, NULL, NULL,
+       napi_enumerable, NULL},
+      {"superclassStaticMethod", NULL, superclass_static_method, NULL, NULL,
+       NULL, napi_enumerable | napi_static, NULL},
+      {"superclassOverriddenMethod", NULL, superclass_overridden_method, NULL,
+       NULL, NULL, napi_enumerable, NULL},
+      {"superclassOverriddenStaticMethod", NULL,
+       superclass_overridden_static_method, NULL, NULL, NULL,
+       napi_enumerable | napi_static, NULL},
+      {"superclassChainedMethod", NULL, superclass_chained_method, NULL, NULL,
+       NULL, napi_enumerable, NULL},
+      {"superclassChainedStaticMethod", NULL, superclass_chained_static_method,
+       NULL, NULL, NULL, napi_enumerable | napi_static, NULL}};
+
+  napi_property_descriptor subclass_properties[] = {
+      {"subclassMethod", NULL, subclass_method, NULL, NULL, NULL,
+       napi_enumerable, NULL},
+      {"subclassStaticMethod", NULL, subclass_static_method, NULL, NULL, NULL,
+       napi_enumerable | napi_static, NULL},
+      {"superclassOverriddenMethod", NULL,
+       subclass_superclass_overridden_method, NULL, NULL, NULL, napi_enumerable,
+       NULL},
+      {"superclassOverriddenStaticMethod", NULL,
+       subclass_superclass_overridden_static_method, NULL, NULL, NULL,
+       napi_enumerable | napi_static, NULL},
+      {"superclassChainedMethod", NULL, subclass_superclass_chained_method,
+       NULL, NULL, NULL, napi_enumerable, NULL},
+      {"superclassChainedStaticMethod", NULL,
+       subclass_superclass_chained_static_method, NULL, NULL, NULL,
+       napi_enumerable | napi_static, NULL}};
+
+  NAPI_CALL(env, napi_define_class(env, "Superclass", NAPI_AUTO_LENGTH,
+                                   construct_superclass, NULL,
+                                   sizeof(superclass_properties) /
+                                       sizeof(superclass_properties[0]),
+                                   superclass_properties, &superclass),
+            "Failed to define superclass");
+
+  NAPI_CALL(env,
+            napi_define_class(
+                env, "Subclass", NAPI_AUTO_LENGTH, construct_subclass, NULL,
+                sizeof(subclass_properties) / sizeof(subclass_properties[0]),
+                subclass_properties, &subclass),
+            "Failed to define subclass");
+
+  napi_inherits(env, subclass, superclass);
```

Comment:
I think using napi_inherits is confusing as it might make people think it is part of the N-API.  I suggest.  setup_inheritance(